### PR TITLE
Print information after login

### PR DESCRIPTION
**Pull request by csweichel**
## Description
Once login is successful, we now print more details about the login user. It's essentially the same as running "Who am I?" right after the login.

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at cffb2f4</samp>

This pull request enhances the user experience of the local-app `login` and `whoami` commands. It adds more logging and feedback for the `login` command, and simplifies the code for both commands by using the `context` package and a shared helper function.

</details>

## Related Issue(s)

Fixes EXP-886

## How to test
```
gitpod login
```

image

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/local-app/cmd/whoami.go [hunk 1]
```diff
@@ -28,27 +31,37 @@ var whoamiCmd = &cobra.Command{
...
-				OrgID: org.Msg.GetTeam().Id,
-				Host:  gpctx.Host.String(),
-			},
-		}, whoamiOpts.Format, prettyprint.WriterFormatNarrow)
+		return WriteTabular(who, whoamiOpts.Format, prettyprint.WriterFormatNarrow)
 	},
 }
 
+// printWhoami prints information about the currently logged in user
+func whoami(ctx context.Context, client *client.Gitpod, gpctx *config.ConnectionContext) ([]whoamiResult, error) {
```
Comment (filiptronicek) [inline components/local-app/cmd/whoami.go:44]: nit: there's a mismatch between the function name and its doc

File: components/local-app/cmd/login.go [hunk 2]
```diff
@@ -66,7 +66,8 @@ var loginCmd = &cobra.Command{
 
 		err = auth.SetToken(loginOpts.Host, token)
 		if err != nil {
-			slog.Warn("could not write token to keyring, storing in config file instead", "err", err)
+			slog.Debug("could not write token to keyring, storing in config file instead")
```
Comment (filiptronicek) [inline components/local-app/cmd/login.go:69]: Could be worth checking `slog.Level` here to omit duplicate output.

**Review by filiptronicek — APPROVED** [review]
Works like a charm.

https://github.com/gitpod-io/gitpod/pull/19029#discussion_r1385224863 feels like worth addressing here, but otherwise :shipit:.

File: components/local-app/cmd/login.go (hunk 2 above)
Comment (mustard-mh) [reply components/local-app/cmd/login.go:69]: Maybe add a common function

**Pull request merged**